go/systests/team_invite_test.go

```diff
@@ -199,3 +203,99 @@ func TestTeamInviteResetNoKeys(t *testing.T) {
 }
+
+func prooveRooter(g *libkb.GlobalContext, user *smuUser) {
+	runner := client.NewCmdProveRooterRunner(g, user.username)
+	err := runner.Run()
+	require.NoError(user.ctx.t, err)
+}
+
+func TestImpTeamInvite(t *testing.T) {
+	ctx := newSMUContext(t)
+	defer ctx.cleanup()
+
+	// Sign up two users
+	alice := ctx.installKeybaseForUser("alice", 10)
+	alice.signup()
+	divDebug(ctx, "Signed up alice (%s)", alice.username)
+	bob := ctx.installKeybaseForUser("bob", 10)
+	bob.signup()
+	divDebug(ctx, "Signed up bob (%s)", bob.username)
+
+	rooterUser := bob.username + "@rooter"
+	displayName := strings.Join([]string{alice.username, rooterUser}, ",")
+
+	team := alice.lookupImplicitTeam(true /*create*/, displayName, false /*isPublic*/)
+
+	divDebug(ctx, "Created implicit team %s\n", team.ID)
+
+	// TODO: Test chats, but it might be hard since implicit team tlf
+	// name resolution for chat commands needs KBFS running.
+
+	prooveRooter(bob.getPrimaryGlobalContext(), bob)
+
+	G := alice.getPrimaryGlobalContext()
+	kickTeamRekeyd(G, t)
+
+	// Poll for new team name, without the "@rooter"
+	newDisplayName := strings.Join([]string{alice.username, bob.username}, ",")
+
+	var team2 smuImplicitTeam
+	var err error
+	for i := 0; i < 5; i++ {
+		team2, err = alice.lookupImplicitTeamError(false /*create*/, newDisplayName, false /*isPublic*/)
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	require.NoError(t, err)
+	require.Equal(t, team2.ID, team.ID)
+
+	// Lookup by old name should fail
+	_, err = alice.lookupImplicitTeamError(false /*create*/, displayName, false /*isPublic*/)
+	require.Error(t, err)
+}
+
+func TestImpTeamInviteWithConflicts(t *testing.T) {
+	ctx := newSMUContext(t)
+	defer ctx.cleanup()
+
+	// Sign up two users
+	alice := ctx.installKeybaseForUser("alice", 10)
+	alice.signup()
+	divDebug(ctx, "Signed up alice (%s)", alice.username)
+	bob := ctx.installKeybaseForUser("bob", 10)
+	bob.signup()
+	divDebug(ctx, "Signed up bob (%s)", bob.username)
+
+	rooterUser := bob.username + "@rooter"
+	displayNameRooter := strings.Join([]string{alice.username, rooterUser}, ",")
+
+	team := alice.lookupImplicitTeam(true /*create*/, displayNameRooter, false /*isPublic*/)
+
+	divDebug(ctx, "Created implicit team %q -> %s\n", displayNameRooter, team.ID)
+
+	displayNameKeybase := strings.Join([]string{alice.username, bob.username}, ",")
+	team2 := alice.lookupImplicitTeam(true /*create*/, displayNameKeybase, false /*isPublic*/)
+
+	divDebug(ctx, "Created implicit team %q -> %s\n", displayNameKeybase, team2.ID)
+
+	prooveRooter(bob.getPrimaryGlobalContext(), bob)
+
+	G := alice.getPrimaryGlobalContext()
+	kickTeamRekeyd(G, t)
+
+	var err error
+	for i := 0; i < 10; i++ {
+		// Poll until the rooter display name stops resolving
+		_, err = alice.lookupImplicitTeamError(false /*create*/, displayNameRooter, false /*isPublic*/)
+		if err != nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
```

Comment:
this time.Sleep will probably be an issue on CI.  I'd recommend using `waitForTeamChangedGregor` or something like it like in `TestTeamInvite*`...